feat(search): introduced search bar functionality

ovflowd opened the pull request:
## Description

This PR introduces (again) a search box for searching content from the Learn Tab. It indexes in the same way and it uses Elastic Lunr.

The original SearchBox component was copied from https://github.com/nodejs/nodejs.dev/pull/1677 and modified both in functionality to work with the plugin and design to be more responsive.

### Screenshots

<img width="322" alt="image" src="https://user-images.githubusercontent.com/12037269/175005529-6e771312-0644-465b-b375-bb8650c450f2.png">

<img width="529" alt="image" src="https://user-images.githubusercontent.com/12037269/175005594-18c217f9-710f-48a4-9bbf-f7251088005c.png">

<img width="545" alt="image" src="https://user-images.githubusercontent.com/12037269/175005625-a5d7b7f5-f49c-445d-b044-c399104054f2.png">

<img width="585" alt="image" src="https://user-images.githubusercontent.com/12037269/175006614-074e6b29-b30f-4599-901f-02c758fc5594.png">


## Related Issues

Fixes #413

ovflowd commented:
**Note.:** This search box is rendered on all pages. If you folks prefer it to only show on Learn, maybe UX-wise it would fit somewhere else?

benhalverson commented:
Hi @ovflowd Thanks for working on this. I like the global search over the /learn specific one.

I noticed there are horziontal and vertical scrollbars for the search bar divs. 
![searchDefault](https://user-images.githubusercontent.com/7907232/175184427-e52341b6-7f55-4dc5-9d34-4bb414488cf6.PNG)
![expandedSearch](https://user-images.githubusercontent.com/7907232/175184568-941e4f49-273f-4a78-a2e7-7c412f5b1715.png)

ovflowd commented:
Hey @benhalverson quite interesting, let me see if removing the `overflow` styles could work.

Would you mind telling me if you're using Linux/Windows? Cause on macOS I'm not able to reproduce this, nor on Fedora.

ovflowd commented:
Also, thanks for enjoying this approach! I thought that a global approach would be easier for people to notice, and also a version that auto-expands doesn't consume that much real-estate 👀

ovflowd commented:
I introduced some style fixes and improved how the search container resizes dynamically based on your screen size 😄

benhalverson commented:
> Hey @benhalverson quite interesting, let me see if removing the `overflow` styles could work.
> 
> Would you mind telling me if you're using Linux/Windows? Cause on macOS I'm not able to reproduce this, nor on Fedora.
I was testing on Windows 10 with Chrome

ovflowd commented:
> I was testing on Windows 10 with Chrome

Gotcha! Hopefully latest changes fix your issue 🙇

mikeesto commented on src/components/SearchBar/SearchBar.scss:
```diff
@@ -0,0 +1,84 @@
+  height: fit-content;
+  max-height: 90vh;
+  min-height: 5em;
+  overflow-y: scroll;
```
```suggestion
```

For what it's worth, this change looked better for me on Chrome/Safari/Firefox. But I'm also on MacOS 😅 

<img width="1154" alt="Screen Shot 2022-06-24 at 4 13 27 pm" src="https://user-images.githubusercontent.com/21051488/175473953-b51a1075-a953-4e74-9769-3fbe5d1a3e55.png">

ovflowd replied:
Oh, had no idea this was generating an outer scroll-bar 🤯

benhalverson reviewed (approved):
👍🏻

ovflowd commented:
@benhalverson I updated the SearchInput's CSS to match more the design from the Figma's.

![image](https://user-images.githubusercontent.com/12037269/175909606-efee9afd-49a8-4789-82db-34e03534b9f2.png)

I also fixed the results not clearing when collapsing and added a limit of 20 results, so that it doesn't show an infinitude of results.
